pkg/asset/cluster/tfvars.go

```diff
@@ -510,17 +510,38 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			caCert = string(caFile)
 		}
 
-		masters, err := mastersAsset.Machines()
-		if err != nil {
-			return err
+		var masterSpecs []*openstackprovider.OpenstackProviderSpec
+		{
+			masters, err := mastersAsset.Machines()
+			if err != nil {
+				return err
+			}
+
+			for _, master := range masters {
+				masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+			}
 		}
 
-		var masterSpecs []*openstackprovider.OpenstackProviderSpec
-		for _, master := range masters {
-			masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+		var workerSpecs []*openstackprovider.OpenstackProviderSpec
+		{
+			workers, err := workersAsset.MachineSets()
+			if err != nil {
+				return err
+			}
+
+			for _, worker := range workers {
+				workerSpecs = append(workerSpecs, worker.Spec.Template.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+			}
 		}
+
+		var workermpool *openstack.MachinePool
+		if len(installConfig.Config.Compute) > 0 {
+			workermpool = installConfig.Config.Compute[0].Platform.OpenStack
+		}
```

Comment:
This doesn't look right: we're only going to consider a server group defined in the first compute config.

Reply:
install-config's "Compute" slice only accepts one machinepool. I should add a comment.

Reply:
(added a comment)